### PR TITLE
Pie and Donut render errors

### PR DESCRIPTION
**Pull request by paulo-rico**
As per issue [Issue](https://github.com/radzenhq/radzen-blazor/issues/810#issue-1581898242) / [Forum Entry](https://forum.radzen.com/t/radzendonut-series-display-error-when-items-in-collection-are-zero/13007)

[Working demo](http://radzen.delaci.co.uk/)

Checks for instances where sum of data = 0 and draws an empty circle rather than nothing (this was causing rendering errors in the browser where some of the values were NaN).

Whilst looking at this, there were also render errors when drawing data labels. These have been resolved.

Also found an error during testing when Data was set to a null object, i.e. DataItem[] revenue; These will also render as an empty circle rather crashing (crashing, not browser rendering error).

Regards

Paul Ruston

### CODE REVIEW

**Comment by akorchev** [comment]
Hi @paulo-rico,

Thank you for the PR. I think the approach should be different than checking if the sum of all series is 0. It should ignore 0 values altogether (in fact values that are `<= 0`). Also the sum can be zero in other cases (e.g. -1 and 1).

In such edge cases I always check how Highcharts behaves (the most capable JS chart IMHO) . Here is what it does when the sum is 0 but there is one positive value:

```
    series: [{
        name: 'Share',
        data: [
            { name: 'Chrome', y: -1 },
            { name: 'Edge', y: 1 },
        ]
    }]
```

image



The negative series is completely ignored but the positive is still rendered. Here is what it does when all values are 0.


```
    series: [{
        name: 'Share',
        data: [
            { name: 'Chrome', y: 0 },
            { name: 'Edge', y: 0 },
        ]
    }]
```

image

It displays data labels (with arrows which the Radzen Chart does not support) but the chart is just a circle.

And here is what it does when all values are negative:

```
    series: [{
        name: 'Share',
        data: [
            { name: 'Chrome', y: -1 },
            { name: 'Edge', y: -1 },
        ]
    }]
```
image

In short I think that the algorithm should be like this:

1. Filter values that are less than or equal to 0.
2. If the collection is empty - draw the empty circle. 

How does this sound?

**Comment by paulo-rico** [comment]
Hi @akorchev 
Hope this looks better. Negative values had never registered with me for a Pie Chart. I must learn to have a component developer hat on instead of a system developer one 😁

[Demo Site](http://radzen.delaci.co.uk/)

Many Regards

Paul

**Comment by akorchev** [comment]
That looks better! I have one small request - can you move this property to the series that need it (pie, donut) instead of the base class?

**Comment by paulo-rico** [comment]
That's done.

Regards Paul
